api/proto/teleport/legacy/types/types.proto

```diff
@@ -3926,6 +3926,8 @@ message OIDCConnectorSpecV3 {
   bool AllowUnverifiedEmail = 15 [(gogoproto.jsontag) = "allow_unverified_email,omitempty"];
   // UsernameClaim specifies the name of the claim from the OIDC connector to be used as the user's username.
   string UsernameClaim = 16 [(gogoproto.jsontag) = "username_claim,omitempty"];
+  // ForceReauthentication forces the user to reauthenticate when logging in.
+  bool ForceReauthentication = 17 [(gogoproto.jsontag) = "force_reauthentication,omitempty"];
```

Comment:
Should we just expose this as `MaxAge` instead?

Reply:
I wasn't sure, we certainly can but I was just erring on the side of simplicity for the user as no one's asked for this before

Reply:
Yeah, I can see that argument. But sometimes for things as complicated as SSO it makes more sense to expose the field directly rather than creating some abstraction over it.

People familiar with the OIDC flow might know to search for `max_age` but they probably won't know that we hid that option behind something called `force_reauthentication`.

Reply:
I've been trying to think of a simple way to make `max_age` nullable, but I'm not sure how to do that with protobufs. Apparently `optional` is off the table, and `gogoproto.nullable` doesn't seem to change the representation of `max_age`. This matters because the zero-value for `max_age` is a valid value, but a user may not want `max_age` set at all to keep existing behavior.

I toyed with the idea of setting a default value for `max_age` that was very large, but that would technically change behavior silently and be backwards incompatible. Setting `max_age` to even a large value would be more secure than not setting it at all, but I'm not sure how important it is to keep behavior *exactly* the same if a user upgrades Teleport in this case. WDYT @zmb3?

Reply:
Another (ugly) option is to introduce a second field, ie `max_age_enabled` but that seems very unintuitive

Reply:
I think we need to bump the resource version to keep this backwards compatible. So the current version of the connector will not support max_age parameter and v4 will.

Reply:
Fixed with help from @codingllama

Reply:
I figured out a way to make `max_age` nullable, so existing connectors will default to have it set to `nil` and after updating users can set `max_age` to a concrete value if they wish. This new reauth behavior will only take effect when `max_age` is set. Not sure how that's backwards incompatible?

Reply:
Yeah that should be fine then. Just make sure to test it please.